src/main/groovy/util/Node.java

```diff
@@ -537,26 +539,40 @@ private NodeList getByName(String name) {
 
     /**
      * Provides a collection of all the nodes in the tree
-     * using a depth first traversal.
+     * using a depth-first preorder traversal.
      *
      * @return the list of (depth-first) ordered nodes
      */
     public List depthFirst() {
+        return depthFirst(true);
+    }
+
+    /**
+     * Provides a collection of all the nodes in the tree
+     * using a depth-first traversal.
+     *
+     * @param preorder if false, a postorder depth-first traversal will be performed
```

Comment:
That's true, but one could convert the tree to a BST before performing the traversal.  Pretty expensive, but could be done.